File: 3333/3333.cs
public class Solution {
    public int FindTheOriginalTypedStringIi(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(FindTheOriginalTypedStringIi(root.left), FindTheOriginalTypedStringIi(root.right));
    }
}

File: 3333/3333.c
int findTheOriginalTypedStringIi(struct TreeNode* root) {
    if (!root) return 0;
    int l = findTheOriginalTypedStringIi(root->left);
    int r = findTheOriginalTypedStringIi(root->right);
    return 1 + (l > r ? l : r);
}